pallets/asset-index/src/benchmarking.rs

```diff
@@ -110,39 +99,57 @@ benchmarks! {
 			admin_deposit.into(),
 		));
 
+		let index_tokens = AssetIndex::<T>::index_token_balance(&origin_account_id).into();
 		T::PriceFeedBenchmarks::create_feed(origin_account_id.clone(), asset_id).unwrap();
+		assert_ok!(T::Currency::deposit(asset_id, &origin_account_id, units));
 
 		// construct call
 		let call = Call::<T>::deposit(asset_id, units);
 	}: { call.dispatch_bypass_filter(origin)? } verify {
 		let nav = AssetIndex::<T>::nav().unwrap();
 		let deposit_value = T::PriceFeed::get_price(asset_id).unwrap().checked_mul_int(units.into()).unwrap();
-		let received = nav.reciprocal().unwrap().saturating_mul_int(deposit_value).saturating_add(1u128);
-		assert_eq!(AssetIndex::<T>::index_token_balance(&origin_account_id).into(), received + admin_deposit as u128);
+		let received = nav.reciprocal().unwrap().saturating_mul_int(deposit_value);
+
+		// NOTE:
+		//
+		// the result will be 0 or 1
+		//
+		// - 0 for tests
+		// - 1 for benchmarks ( transaction fee )
+		assert!(AssetIndex::<T>::index_token_balance(&origin_account_id).into() - (index_tokens + received) < 2);
 	}
 
-	remove_asset {
-		let asset_id :T::AssetId =  T::try_convert(2u8).unwrap();
-		let origin = T::AdminOrigin::successful_origin();
-		let units: u32 = 100;
-		let amount = 500u32.into();
-
-		assert_ok!(AssetIndex::<T>::add_asset(
-			origin.clone(),
-			asset_id,
-			units.into(),
-			MultiLocation::Null,
-			amount,
-		));
-
-		// ensure
-		assert_eq!(T::IndexToken::total_balance(&Default::default()), 500u32.into());
-
-		// construct remove call
-		let call = Call::<T>::remove_asset(asset_id, units.into(), None);
-	}: { call.dispatch_bypass_filter(origin.clone())? } verify {
-		assert_eq!(T::IndexToken::total_balance(&Default::default()), 0u32.into());
-	}
+	// TODO:
+	//
+	// This extrinsic requires `remote-asset-manager`
+	//
+	// ----
+	//
+	// remove_asset {
+	// 	let asset_id =  T::try_convert(2u8).unwrap();
+	// 	let units = 100_u32.into();
+	// 	let amount = 1_000u32.into();
+	// 	let origin = T::AdminOrigin::successful_origin();
+	// 	let origin_account_id = T::AdminOrigin::ensure_origin(origin.clone()).unwrap();
+	// 	let receiver = whitelisted_account::<T>("receiver", 0);
+	//
+	// 	// create liquid assets
+	// 	assert_ok!(<AssetIndex<T>>::add_asset(
+	// 		origin.clone(),
+	// 		asset_id,
+	// 		units,
+	// 		MultiLocation::Null,
+	// 		amount
+	// 	));
+	//
+	// 	// create price feed
+	// 	T::PriceFeedBenchmarks::create_feed(origin_account_id.clone(), asset_id).unwrap();
+	//
+	// 	// construct call
+	// 	let call = Call::<T>::remove_asset(asset_id, units, Some(receiver));
```

Comment:
good catch with all the xcm dependent calls.
They can't be benchmarked like that in any case so we can remove them.
the weights for these calls will come from the upcoming xcm benchmarking suite in polkadot.